### ISSUE TITLE
OCDS License (Was governance process)

### ISSUE THREAD

**Issue by timgdavies** [open-contracting/standard#272]
**Update:** A revised governance draft [was posted here](http://standard.open-contracting.org/1.0-adding-governance/en/support/governance/) on 15th Feb, and will shortly be reviewed by OCP for final adoption.

The outstanding issue of license will be covered by further discussions in this thread. 

-------

The Open Contracting Data Standard (OCDS) has many stakeholders: governments (procuring entities, monitoring & oversight authorities, project managers and policy makers), the private sector, and civil society organizations. The needs and interests of these stakeholders, as publishers and users of the data, are varied. As OCDS develops over time, with updated versions and new features, it is important that a diverse group of stakeholders are engaged in the process.

We have prepared a short document to set out a **proposed process and governance model for creating version 1.1** of OCDS for consultation ([GDocs](https://docs.google.com/document/d/1Gy1tCYPjqPzoR58sI2BTZYuk3kNCck-sf6N0IB-OKic/edit#) / [PDF](https://github.com/open-contracting/standard/files/49917/OCDS.Governance.Proposal.-.December.2015.pdf))

## Share your feedback
We would welcome your comments on the proposed governance and process **by 5th January 2016**

You can [comment directly on the document in Google Docs format here](https://docs.google.com/document/d/1Gy1tCYPjqPzoR58sI2BTZYuk3kNCck-sf6N0IB-OKic/edit#heading=h.a0713pulvneo), or add your responses in the thread below. 

## Process summary
The revision process is summarised in the graphic below. 

![image](https://cloud.githubusercontent.com/assets/342624/11536681/6af0d754-9911-11e5-84a6-e5f349314ee5.png)

The proposal envisages adoption of the [Open Stand principles](https://open-stand.org/about-us/principles/), the creation of a Governance Working Group as an oversight and appeal body for the revision process and the recruitment of invited peer-reviewers to ensure a diversity of in-depth stakeholders input into the open peer review process. 

The document contains a worked example of steps and timeline for the proposed 1.1 process. 

## Where next
Following feedback on this draft, and based on that feedback, we anticipate that:

* We will create a final revision control document, to include as part of future documentation. This will include checklists for each stage of the revision process.
* We will start the 1.1 revision process.
* Appropriate groups (peer-reviewers, standard governance working group) will be established.

## Engage

* By commenting on this thread;
* By adding comments directly to the Google doc;

Comments in support of the proposal (+1), or with blocking objections (explicitly stated) are encouraged.

**Comment by mlenczner** [comment]
Seems fine. As someone who's interested in applying this standard in another country than the UK, I'm interested in the next steps.

How will the following step take place: "Appropriate groups (peer-reviewers, standard governance working group) will be established."

No need to respond here. Just noting my interest in that question.

**Comment by timgdavies** [comment]
Thanks @mlenczner. 

The proposal suggests that the **Standard Governance Working Group** will be made up of representatives from OCP staff, the multi-stakeholder advisory board, and the technical team - and so I anticipate it will be up to these groups to nominate their representatives, subject to the principles of ensuring balance of stakeholder representation. 

On peer reviewers - I'm anticipating we'll put out an open call, and then, in consultation with the Standard Governance Working Group may reach out to particular stakeholder groups under-represented in open call responses. There is no limit on the number of peer-reviewers we might have from different stakeholder groups, but we do want to make sure we specifically ensure a breadth of stakeholder groups are represented, and that there is some degree of balance to the group.

**Comment by timgdavies** [comment]
From offline comments, the current IP statement in the document has been highlighted as potentially problematic. It currently reads:

> The Open Contracting Data Standard is the intellectual property of the Open Contracting Partnership.  The schema is provided under the Creative Commons Attribution-NoDerivatives 4.0 International license, with accompanying documentation under a Creative Commons Attribution 4.0 International license. 
>
> Contributors to the standard agree to transfer any copyright in their contributions to the Open Contracting Partnership, in order that it is held in trust as part of the standard.

This implies a **Copyright Assignment** style Contributor Agreement. The alternative to this is a **Copyright License** style Contributor Agreement, in which contributors maintain copyright, but grant a irrevocable license to the OCP for those contributions.

Groups such as [W3C use a Copyright License approach](https://www.w3.org/community/about/agreements/cla/) whereas [the Free Software Foundation uses Copyright Assignment](https://www.gnu.org/licenses/why-assign.en.html). 

Our view was that Copyright Assignment model would be simpler to manage, and given the nature of contributions to the standard will generally involve small additional field name suggestions and elements of documentation, we did not envisage anyone having a strong interest in retaining individual copyright in these. 

However, feedback on this is welcome. Would a Copyright Assignment based CLA present any practical challenges to organisations who would otherwise be able to contribute to the core standard? 

Are there any strong reasons for us to prefer either copyright licensing or assignment models?

**Comment by CountCulture** [comment]
What's the thinking behind the NoDerivatives restriction?

**Comment by timgdavies** [comment]
The NoDerivatives was based on [W3C practice](http://www.w3.org/Consortium/Legal/2015/doc-license) and the idea that the correct way to modify the standard is via extensions, or via engaging in the revision process - rather than creating a forked version.

**Comment by CountCulture** [comment]
OK, but does ND prohibit reuse of part of the standard?
In the W3C standard it says: "To facilitate implementation of the technical specifications set forth in this document, anyone may prepare and distribute derivative works and portions of this document in software, in supporting materials accompanying software, and in documentation of software, PROVIDED that all such works include the notice below. HOWEVER, the publication of derivative works of this document for use as a technical specification is expressly prohibited."

Not a big issue for me, just trying to understand...

**Comment by timgdavies** [comment]
That's a good point:

Just re-reading the human language version of ND terms which states:

>NoDerivatives — If you remix, transform, or build upon the material, you may not distribute the modified material.

I have a feeling we might want that to apply to the ```schema``` to maintain integrity of the standard, but we wouldn't want anyone to be prohibited from remixing the docs and distributing that. 

Perhaps a custom statement such as the W3C prohibition would be more appropriate - although that takes us down the road of a custom license, rather than something off the shelf unless there is an alternative we've not found here.

**Comment by jpmckinney** [comment]
I skimmed the document (I read the earlier version - I think most of my comments are integrated into this one). I prefer BY over BY-ND. I think the ND part will never be enforced, and it just creates confusion as to how you may extend, reuse, create profiles for, etc. the standard.

**Comment by bill-anderson** [comment]
Regarding IP, is OCP a legal entity? (As an aside the website -even if only small print - could be clearer on this point.) IATI isn't, which means it cannot own IP.

**Comment by timgdavies** [comment]
We've passed the formal deadline for responses to the draft, but it will be a few more days before we review and respond to all the comments - so if any other thoughts those would be really welcome before end of Friday.

**Comment by timgdavies** [comment]
@bill-anderson Right now OCP is a project hosted by a Fiscal Sponsor. We're working out what the small print will need to say on the ownership / licensing of the standard itself.

**Comment by markbrough** [comment]
I support @jpmckinney and would also suggest going for a CC-BY license rather than CC-BY-ND
- It seems getting people to publish to the standard should be the priority rather than worrying about the extent to which people are complying with it at this stage.
- Presumably if someone tracks the standard closely but deviates in a couple of ways (e.g. different codelists, or including some other data they need to add for legal or other reasons) then they could fall foul of the ND clause. 
- There are likely other strong incentives for organisations to closely track the predominant / official standard (as they're trying to publish data in a standard format that others can use); legal restrictions don't seem necessary.
- OCDS is at a much less mature stage than most W3C standards, and there will inevitably be ways in which the standard will need to be modified or adapted as real world implementation starts to take place. I can see this being particularly problematic with some of the [closed codelists](http://ocds.open-contracting.org/standard/r/1__0__0/en/schema/codelists/#method), for example.
- I struggle to see how ND would have improved the quality of the data in IATI but can absolutely see it holding up implementation.
- [CC-BY-ND does not satisfy the open definition](http://opendefinition.org/licenses/nonconformant/)…

I also see some truth in [this comment](https://twitter.com/OPENGOVHULK/status/688069900179386368) - creating legal restrictions on how the standard can be used or adapted could cause some organisations problems in terms of implementation if they have to get the lawyers involved. There should be no doubt at all that OCDS does not lock you in.

**Comment by greggish** [comment]
What about CC-BY-SA? 

This would ensure that adaptations remain open source. However, perhaps this does not actually address the concern of schema fragmentation. OTOH, is there a reason _not_ to require Share-Alike?

(I'm not familiar with this domain or invested in this issue, just interested in this challenge of governance / licensing specs, and curious about folks' perspectives on this.)

**Comment by timgdavies** [comment]
So - just to check the starting point: the thing we're discussing, and that is licensed CC-BY-ND at the moment is the **schema and documentation**, and the license is intended to apply only to **distribution** of derivatives. 

This means:
- Users _can_ create JSON files with whatever structure they like, breaking with the schema as they see fit;
- They can license those files with whatever license they want (though for this [we recommend](http://ocds.open-contracting.org/standard/r/1__0__0/en/implementation/publication_patterns/#licensing) use of PD Dedication of CC-BY;

So in @markbrough's examples above, you would currently only fall foul of the license if you created a schema for your customised data, and published that, not publishing the data itself. 

On the reasons for the current approach, the primary concern was to find a framework for stewardship of the standard that protects against fragmentation from, for example, industry forks focussed on their specific business cases that bypass a collaborative standards process. For example, if someone wanted to use a different set of closed codelist codes, they can in their own data, but they can't fork the schema and produce an alternative standard with different codelists. Instead, they would be encouraged to engage with the main standards process. This kind of issue is more a concern in the early stages before the kinds of other strong incentives Mark cites kick in from the network effects of a core standard. 

Given the focus on avoiding fragmentation, @greggish's suggestion of Share Alike would probably not fit as well. 

All this said, going back to the [W3C document licensing policy](https://www.w3.org/Consortium/Legal/2015/doc-license) that led us towards CC-BY-ND, suggests a more nuanced approach. The key terms there state:

> No right to create modifications or derivatives of W3C documents is granted pursuant to this license, except as follows: To facilitate implementation of the technical specifications set forth in this document, anyone may prepare and distribute derivative works and portions of this document in software, in supporting materials accompanying software, and in documentation of software, PROVIDED that all such works include the notice below. HOWEVER, the publication of derivative works of this document for use as a technical specification is expressly prohibited.

This does suggest to me that our current ND is too broad, but that there may still be reasons not to simple remove the restriction entirely. 

Thoughts?

**Comment by jpmckinney** [comment]
My earlier opinion is the same now:

> I think the ND part will never be enforced, and it just creates confusion as to how you may extend, reuse, create profiles for, etc. the standard.

An important question is whether having an ND license actually prevents or mitigates fragmentation. I don't think it does. The risk of fragmentation isn't mitigated by some words in a license file. It's mitigated through promotion, active engagement with users, responsiveness to new needs, etc. Open311 didn't suffer from fragmentation due to any problems with its license. It underwent some fragmentation because the standard's custodians were otherwise occupied and unable to spend the time necessary to hold the stakeholders together.

If a standard provides value, but the official channels are unresponsive, slow, or whatever else, then people will fork it. And if you try to enforce the license, they will find ways around the licensing issues, often in ways that make the fork more divergent (e.g. renaming fields or cutting features just to avoid being too close to the standard from which it is derived).

So, in any world, I don't see how ND protects from fragmentation of a standard.

**Comment by bill-anderson** [comment]
As Tim argues this is about the integrity of the standard, not the data. Forking a standard means that you now have two competing standards. The whole point of building a standard (and taking the pain to get it right) is to avoid derivatives that complicate compatibility and interoperability. 

The IATI standard remains far from perfect but the de facto ND status it has means everyone fights together to improve it.

**Comment by markbrough** [comment]
I think it would be a very welcome thing if industry forks started using something very close to Open Contracting as a data exchange language. If they had some slight deviations from OCDS (to cover some pieces of information that may or may not be of wider interest), it would be a good thing for them to publish the new schema or codelists rather than be legally required not to publicly disclose the new schema or codelists they are using.

IATI schema and codelists are [CC-BY](https://github.com/IATI/IATI-Guidance/blob/master/en/license.rst) or [MIT](https://github.com/IATI/IATI-Codelists-NonEmbedded/blob/master/LICENSE). This is good; a couple of weeks ago I needed to have French-language codelists for a project I'm working on. I was able both to create a derivative work of the code and codelists and (just now) [shared the content](https://github.com/markbrough/IATI-Codelists-NonEmbedded/commit/6c9ee9c88ec7af9d280f43787400dff136cd8189#diff-dd45cb458a9ba8f4c80960b94d8e1ecb), in a way that is compliant with the existing codelist schema and pretty closely tracks the existing (and [other proposed](https://github.com/IATI/IATI-Codelists-NonEmbedded/pull/51)) code. I'm not sure why you would want to discourage that, and indeed, Github makes creating and sharing derivatives especially easy -- currently there are 17 forks of this repository! IATI's de jure status of CC-BY has not prevented it from being (as @bill-anderson points out) sort of de facto CC-BY-ND as there is no big incentive for publishing in some significantly different / competing fork of the standard, and official channels are pretty responsive.

**Comment by timgdavies** [comment]
Thanks all for the discussion here: really really useful (albeit I think we all recognise we're mostly dealing with hypothetical edge cases, given the limited bite that license terms are actually having on behaviours in practice...).

So - dropping the Creative Commons ND seems to have been strongly argued for. This leaves two options:
1. Simply use MIT / CC-BY
2. Create a custom license based on W3C Document License 

The second option would allow derivatives of the kind Mark describes, but with the caveat that _"publication of derivative works of this document for use as a technical specification is expressly prohibited"_. 

We do have the extensions mechanism for OCDS (i.e. if you extend the standard, publish as JSON Patch for the schema), and I would much prefer to see this route for sector-specific modifications developed instead of seeing multiple schema published.

**Comment by jpmckinney** [comment]
The advantages of MIT and CC-BY are that they are well-known, popular, clear licenses. With W3C (and any derivative we adopt), people need to carefully read and interpret the license. CC-BY 4.0 is a carefully researched international license; I don't know if we'll meet that. Another pro for MIT or CC-BY is that they are open licenses, which may be important in terms of public perception.

Of course, there could be some significant advantages to a custom license that outweigh the disadvantages. Prohibiting "publication of derivative works for use as a technical specification" is an advantage, but how significant?

I understand why the W3C has that clause, because the W3C's primary activity is authoring technical specifications, often in environments in which there are competing standards. The Open Contracting Secretariat is not the same, working on one specification, in an noncompetitive environment. 
So, I don't consider the risk of derivative works for use as technical specifications to be very high, meaning the advantage is not great.

Anyhow, my preferences are not as strong between these two options.

**Comment by markbrough** [comment]
> publication of derivative works of this document for use as a technical specification is expressly prohibited

Still don't see these kinds of legal barriers as being ultimately advantageous. In the [industry example I gave above](https://github.com/open-contracting/standard/issues/272#issuecomment-172518101), I'm not sure why you would want to prohibit them from publishing the schema they've used for data exchange. Arguably my IATI codelists translation example would also fall foul of such a clause. I think the above comment re Open311 is pretty solid - if the processes around the core standard are good, people will generally work with them; if they're slow or unresponsive, people will find a way to work around them. It seems something like MIT is the best way to avoid stifling innovation; and engagement with the community of practice is the best way of avoiding forks that compete with the core branch.

I would also argue against using a custom license - license proliferation is not a healthy thing, and if you would like attract large-ish organisations to contribute / take up OCDS then it's going to be much easier for them to do so under a well-known license than under some custom license which they may have to get the lawyers to look over first… Licenses may not matter much to us, but will matter to them. So I would go for something like MIT or CC-BY.

**Comment by jpmckinney** [comment]
@markbrough +1

**Comment by timgdavies** [comment]
I see the restriction as a protection in the edge-case against adversarial actors, and if modelled on W3C clauses which are widely used, unlikely to cause any extra legal barriers in the day-to-day. 

Procurement data is a different landscape from the aid data landscape, or indeed the 311 civic issue reporting landscape - and one in which there isn't always as collaborative a community of stakeholders as you will find in either of these fields. 

In the long-run we're more likely with OCDS to see vendors asked by government clients to implement OCDS, and in these cases, there may be stronger incentives on them to create their own forks and to push these as alternatives to a multi-stakeholder developed standard. The interests of these vendors will not always be aligned with the public interest goals of OCDS. 

In the everyday case, engagement with community is absolutely core to avoiding fragmentation. But in these adversarial cases I think it is not unreasonable to maintain some defensive tools in the form of the license.

**Comment by jpmckinney** [comment]
@timgdavies :+1: considering some vendors have a lot of government clients.

**Comment by markbrough** [comment]
I don't see how the clause would provide any significant protection against adversarial actors. If a vendor wanted to do something like Open Contracting but different (why?), they could make up their own entirely different standard in order to avoid it being seen as derivative. (Incidentally, in the aid data world there are also large vendors with significant market power; I don't see the two cases being entirely dissimilar.)

OCP may choose not to enforce the non-derivative clause in the less adversarial cases I cited above, but that kind of inconsistent / arbitrary choice to enforce or not to enforce on a case-by-case basis does not provide a helpful or stable environment for software development.

**Comment by bill-anderson** [comment]
On a more philosophical note, IATI tackled this as a pragmatic political problem - this played both to IATI's  strength (getting political buy-in for a multi-stakeholder initiative) and its weakness (technically sloppy). OCDS, I believe, should find a better balance, but it is a balance nevertheless. Technical purity  is not what will build traction and usage.

**Comment by timgdavies** [comment]
So - talking only about the narrow W3C style ND clause here ("publication of derivative works of this document for use as a technical specification is expressly prohibited"), without this, I understand that the following is possible:
- Governments ask vendors to provide OCDS compliant output from their systems;
- Vendors want to push a particular data structure they use, over one that has been agreed through OCDS multi-stakeholder process (various reasons 'why', from minimising effort mapping to common framework, through to actively trying to restrict what gets released / maintain existing vendor lock in by customising version of the standard around that);
- Vendor forks standard, but presents this to government as 'OCDS compliant';

Having the W3C style ND clause would offer one means of protection, as it offers move to (a) directly fight actions of adversarial actor; (b) highlight potential risk to government of using this forked standard as opposed to the multi-stakeholder agreed standard. 

There is no proposal here to apply the clause selectively: the narrower W3C style clause could be applied uniformly to everyone. Essentially it's a social contract term: with it, working with OCDS as a standard involves agreeing to pursue adaptations through the multi-stakeholder process. 

Again - as I understand the narrow W3C term, publishing a patched version of the schema to help people validate your own extended data (e.g. your codelist example above) is entirely allowed. It is only promoting this for others to adopt as a formal technical spec independent of the standards main governance process which would be restricted. 

(*NB: I'm still very open minded on this all - but do want to check we get this decision right and test the arguments...)

**Comment by markbrough** [comment]
I think the example @timgdavies provides is helpful.

In the case where a vendor falsely / fraudulently claims that what they are doing is "OCDS compliant", I think there are a couple of options:
1. the individuals monitoring vendors' implementation of these standards could be given tools to validate the data themselves, if they want to check the quality of the data produced by the vendor.
2. separate the use of Open Contracting trademarks from the standard itself. So, OCP retains rights to the Open Contracting logo, etc, and can provide conditions for when people are allowed to stick the Open Contracting badge on their product. That does not affect the core data standard itself.

The W3C style ND license (the particular clause cited above) does appear to me to prohibit things like making a translation of codelists that others might (reasonably) then want to use, or a vendor making some changes to the standard to convey data important to their sector and then publishing technical documentation explaining what they did. You could create a custom clause which provides various exceptions to narrow the scope of prohibited activity, but you're again facing the problems associated with custom licensing (i.e. the lawyers have to get involved).

I agree with @timgdavies' idea of a social contract whereby people commit to engaging collaboratively, but I think making that social contract legally enforceable / enforced is both unnecessary (it has limited benefits) and a bad idea (it has some clear costs).

**Comment by markbrough** [comment]
Actually, reflecting on this again, the ND clause probably wouldn't provide much protection against bogus implementations of the standard as it would only kick in if a vendor provided technical documentation, not just the data. That seems to be unlikely if a vendor is really providing a bogus implementation (they can just refuse to distribute the documentation). Also, presumably even without trademarks etc, fraud may cover cases where someone claims to be implementing OCDS but actually isn't.

**Comment by mk270** [comment]
The ND clause seems to prohibit the creation of an F# type provider based on automatic reflection of codes and structures programmatically extracted from standard but subject to some kind of trivial but necessary transformation. I don't see why you'd want to put this kind of use-case in legal doubt.

**Comment by timgdavies** [comment]
Thank you to everyone for the contributions to this thread.

A draft of the updated governance document, based on feedback is available at http://standard.open-contracting.org/1.0-adding-governance/en/support/governance/ and is going for final review from the OCP.

This does not yet address in full the license issue, although we have an updated proposal on that based on all the comments here which I'll share in a moment, and will rename this thread to continue the license discussion.

**Comment by timgdavies** [comment]
We recognise the need to update the license for the OCDS schema from Creative Commons 4.0 BY-ND - as the no-derivatives clause here goes beyond the intended purpose of protecting the integrity of the schema. 

We have looked at a variety of license options, threats to the standard, and ways of dealing with this (including community pressure, and rights over the Open Contracting name, alongside license) and have identified the Apache 2.0 License as an appropriate alternative.

The reasons for choosing this include the presence of clear clauses about explicit statements when modifications have been made, and the careful restriction on use of 'product names' / 'trademarks' which should help protect the Open Contracting name, whilst permitting use and modification of the schema. 

![screen shot 2016-02-15 at 18 24 40](https://cloud.githubusercontent.com/assets/342624/13057039/638ca2be-d411-11e5-89b4-0b01357c6b3c.png)

(Summary graphic from http://choosealicense.com/licenses/) 

To change the license, we propose to:

(a) Leave this open for comment for a period of at least 10 days;

(b) If there is rough consensus, submit this as the first issue for consideration by the "standard governance working group" which is to be established by OCP and to be "made up of representatives from OCP staff, the multi-stakeholder advisory board, and the technical team".  

Hopefully that group will be established in the next 3 - 6 weeks;

(c) Following approval of a change by that group (or, if needs be, following further iterations), go to all those who have contributed to date to get their agreement to the new license;

This proposal is based on the use of a **copyright assignment** contributor agreement, as proposed in the governance document. 

Thoughts and views?

**Comment by jpmckinney** [comment]
Assuming the Apache 2.0 License applies effectively to documentation (I only know of its use for software), then it seems reasonable to me.

**Comment by markbrough** [comment]
@timgdavies using the Apache License seems like a very sensible proposal to me. 

Thanks for taking into account suggestions in this thread, looking into different licenses and for proposing this alternative!

**Comment by mk270** [comment]
@timgdavies the use of Apache licensing overcomes my objections with invigorating brutality, thank you!

Echoing @jpmckinney 's point, licences designed for code, for data, or for prose text tend not to work well outside their anticipated domain of applicability, so I'd certainly be interested in knowing how well the Apache licence works as applied to text - is there any existing literature on this?

**Comment by timgdavies** [comment]
Thank you to everyone for your input on license, and apologies for the long delay in sharing an update. The Open Contracting Partnership has also sought wider input into the licensing issue, with advice from licensing experts in industry.

A record of the issues [considered are in this discussion note](https://docs.google.com/document/d/19hOemLBymPLl7WnNYJ_N3zCCnaJvBBlHm55nCfONH_k/edit#heading=h.uqp3wt1f5sbd), which supports the earlier discussions in this thread to switch to use of the Apache 2.0 license.

We have now applied the updated license to the standard, and added a contributor agreement, based on copyright assignment as discussed in the earlier governance documents, to the main repository. 

We looked carefully at whether the Apache 2.0 license could be applied to documentation as well as code, and found it was compatible with documentation of a schema. 

This change hopefully resolves the outstanding blocking issues for the new governance process, as consulted on at the end of last year / early this year, to go live, and so that is also now updated in the standard documentation.
